core/src/main/scala/cats/syntax/coproduct.scala

```diff
@@ -0,0 +1,13 @@
+package cats
+package syntax
+
+import cats.data.Coproduct
+
+trait CoproductSyntax {
+  implicit def coproductSyntax[F[_], A](a: F[A]): CoproductOps[F, A] = new CoproductOps(a)
+}
+
+class CoproductOps[F[_], A](val a: F[A]) extends AnyVal {
```

Comment:
Minor: let's go ahead and make this class `final` to be consistent with other `Ops` classes.